sycl/test/extensions/sub_group_mask.cpp

```diff
@@ -1,85 +1,137 @@
+  {
+    auto mask = sycl::detail::Builder::createSubGroupMask<mask_type>(0, 32);
+    const auto const_mask =
+        sycl::detail::Builder::createSubGroupMask<mask_type>(0, 32);
+
+    static_assert(
+        std::is_same_v<decltype(mask[sycl::id(0)]), mask_type::reference>);
+    static_assert(std::is_same_v<decltype(const_mask[sycl::id(0)]), bool>);
+
+    static_assert(std::is_same_v<decltype(const_mask.test(sycl::id(0))), bool>);
+
+    static_assert(std::is_same_v<decltype(const_mask.any()), bool>);
+    static_assert(std::is_same_v<decltype(const_mask.all()), bool>);
+    static_assert(std::is_same_v<decltype(const_mask.none()), bool>);
+
+    static_assert(std::is_same_v<decltype(const_mask.count()), std::uint32_t>);
+    static_assert(std::is_same_v<decltype(const_mask.size()), std::uint32_t>);
+
+    static_assert(std::is_same_v<decltype(const_mask.find_low()), sycl::id<1>>);
+    static_assert(
+        std::is_same_v<decltype(const_mask.find_high()), sycl::id<1>>);
+
+    int bits_i = 0;
+    unsigned long long bits_ull = 0;
+    sycl::marray<char, 3> bits_mc(0);
+    sycl::marray<unsigned short, 12> bits_ms(0);
+
+    mask.insert_bits(bits_i);
+    mask.insert_bits(bits_ull, sycl::id(0));
+    mask.insert_bits(bits_mc);
+    mask.insert_bits(bits_ms, sycl::id(0));
+
+    const_mask.extract_bits(bits_i);
+    const_mask.extract_bits(bits_ull, sycl::id(0));
+    const_mask.extract_bits(bits_mc);
+    const_mask.extract_bits(bits_ms, sycl::id(0));
+
+    mask.set();
+    mask.set(sycl::id(0));
+    mask.set(sycl::id(0), false);
+
+    mask.reset();
+    mask.reset(sycl::id(0));
+
+    mask.reset_low();
+    mask.reset_high();
+
+    mask.flip();
+    mask.flip(sycl::id(0));
+
+    static_assert(std::is_same_v<decltype(const_mask == mask), bool>);
+    static_assert(std::is_same_v<decltype(const_mask != mask), bool>);
+
+    static_assert(
+        std::is_same_v<decltype(mask &= const_mask), mask_reference_type>);
+    static_assert(
+        std::is_same_v<decltype(mask |= const_mask), mask_reference_type>);
+    static_assert(
+        std::is_same_v<decltype(mask ^= const_mask), mask_reference_type>);
+    static_assert(std::is_same_v<decltype(mask <<= 3u), mask_reference_type>);
+    static_assert(std::is_same_v<decltype(mask >>= 3u), mask_reference_type>);
+
+    static_assert(std::is_same_v<decltype(~const_mask), mask_type>);
+    static_assert(std::is_same_v<decltype(const_mask << 3u), mask_type>);
+    static_assert(std::is_same_v<decltype(const_mask >> 3u), mask_type>);
+
+    static_assert(std::is_same_v<decltype(const_mask & mask), mask_type>);
+    static_assert(std::is_same_v<decltype(const_mask | mask), mask_type>);
+    static_assert(std::is_same_v<decltype(const_mask ^ mask), mask_type>);
+  }
+
+// sycl_ext_oneapi_sub_group_mask rev.2
+#if SYCL_EXT_ONEAPI_SUB_GROUP_MASK >= 2
+  {
+    // sub_group_mask()
+    mask_type mask;
+    assert(mask.none() && mask_type::max_bits == mask.size());
```

Comment:
I wonder if for these we should also check them against the subgroup max local size on the host.

Reply:
Done [0a34335](https://github.com/intel/llvm/pull/11195/commits/0a34335d860a39ac9e50994272d5e271b0f9ef38)